Install/Test/CMakeLists.txt

```diff
@@ -0,0 +1,188 @@
+# cmake 3.18+ to have the ARCHIVE_EXTRACT sub-command for files
+cmake_minimum_required(VERSION 3.18)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+include(FetchContent)
+
+project(TestInstall)
+
+FetchContent_Declare(cmake-extensions
+    GIT_REPOSITORY https://github.com/BabylonJS/CMakeExtensions.git
+    GIT_TAG 366dc4a84fb20f4060d97e89948c343e74c51fc3)
+FetchContent_MakeAvailable(cmake-extensions)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# __cplusplus value is not in sync with language version used. MVSC needs this flag to update it accordingly
+# https://gitlab.kitware.com/cmake/cmake/-/issues/18837
+if (MSVC)
+    add_compile_options(/Zc:__cplusplus)
+endif()
+
+if(NAPI_JAVASCRIPT_ENGINE STREQUAL "")
+    message("No JS Engine provided. Defaulting to Chakra.")
+    set(NAPI_JAVASCRIPT_ENGINE "Chakra")
+endif()
+
+# executable stub
+
+set(APPS_DIR "../../Apps")
+set(UNITTESTS_DIR "${APPS_DIR}/UnitTests")
+set(LOCAL_SCRIPTS "${UNITTESTS_DIR}/Scripts/tests.js")
+
+set(NPM_SCRIPTS
+    "${APPS_DIR}/node_modules/babylonjs/babylon.max.js"
+    "${APPS_DIR}/node_modules/babylonjs/babylon.max.js.map"
+    "${APPS_DIR}/node_modules/babylonjs-materials/babylonjs.materials.js"
+    "${APPS_DIR}/node_modules/babylonjs-materials/babylonjs.materials.js.map"
+    "${APPS_DIR}/node_modules/chai/chai.js"
+    "${APPS_DIR}/node_modules/mocha/mocha.js")
+
+set(HEADERS "${UNITTESTS_DIR}/Shared/Tests.h")
+
+if(WIN32)
+    set(TEST_INSTALL_APP "${UNITTESTS_DIR}/Win32/App.cpp")
+else()
+	message(FATAL_ERROR "Unrecognized platform: ${CMAKE_SYSTEM_NAME}")
+endif()
+
+# Nuget/JSI
+# TODO: move it to generic cmake
+set(NUGET_PATH "${BINARY_DIR}/NuGet")
+set(V8JSI_VERSION "0.64.33")
+set(CPU_ARCH "x64")
+set(PLATFORM_FOLDER "win32")
+set(V8JSI_PACKAGE_PATH "${NUGET_PATH}/packages/ReactNative.V8Jsi.Windows.${V8JSI_VERSION}")
+set(V8JSI_LIB_PATH_RELEASE "${V8JSI_PACKAGE_PATH}/lib/${PLATFORM_FOLDER}/Release/${CPU_ARCH}/")
+
+if(NAPI_JAVASCRIPT_ENGINE STREQUAL "JSI")
+    set(JSI_CPP "${V8JSI_PACKAGE_PATH}/build/native/jsi/jsi/jsi.cpp")
+elseif(NAPI_JAVASCRIPT_ENGINE STREQUAL "V8")
```

Comment:
I wonder if it makes sense for the install to include these v8 specific files so that consumers don't have to include them again? We already have the files during the build.

Reply:
Can this be error prone if mixing different v8/jsi versions at some point?

Reply:
Probably, neither contracts are ABI-safe nor backwards compatible.